staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller_reconcile.go

```diff
@@ -501,13 +502,20 @@ func (c *policyController) latestPolicyData() []policyData {
 				if definitionInfo.lastReconciledValue.Spec.ParamKind != nil {
 					hasParam = true
 				}
+				matchConditions := definitionInfo.lastReconciledValue.Spec.MatchConditions
+				matchExpressionAccessors := make([]cel.ExpressionAccessor, len(matchConditions))
+				for i, _ := range matchConditions {
+					matchExpressionAccessors[i] = (*matchconditions.MatchCondition)(&matchConditions[i])
+				}
 				optionalVars := cel.OptionalVariableDeclarations{HasParams: hasParam, HasAuthorizer: true}
 				expressionOptionalVars := cel.OptionalVariableDeclarations{HasParams: hasParam, HasAuthorizer: false}
+				failurePolicy := convertv1alpha1FailurePolicyTypeTov1FailurePolicyType(definitionInfo.lastReconciledValue.Spec.FailurePolicy)
 				bindingInfo.validator = c.newValidator(
 					c.filterCompiler.Compile(convertv1alpha1Validations(definitionInfo.lastReconciledValue.Spec.Validations), optionalVars, celconfig.PerCallLimit),
+					matchconditions.NewMatcher(c.filterCompiler.Compile(matchExpressionAccessors, optionalVars, celconfig.PerCallLimit), c.authz, failurePolicy, "validatingadmissionpolicy", definitionInfo.lastReconciledValue.Name),
```

Comment:
does `c.filterCompiler.Compile` or `NewMatcher` or the `Match()` impl do appreciable work even if we have no conditions? should we only pass in a matcher if we actually have conditions to match, and skip the `Match()` call if that's nil?

Reply:
Per below, I think this will save on evaluation in the near term (long term we should cache operations like casting the input object as unstructured).

Reply:
If we don't have any matchConditions, skipping the matcher construction here and tolerating nil below is easy to understand for now. I'd probably recommend that

Reply:
added